src/lib/Microsoft.Health.Fhir.R4.Ingest/Service/R4FhirImportService.cs

```diff
@@ -72,26 +72,66 @@ public virtual async Task<string> SaveObservationAsync(ILookupTemplate<IFhirTemp
                 existingObservation = await GetObservationFromServerAsync(identifier).ConfigureAwait(false);
             }
 
-            Model.Observation result;
+            Model.Observation result = null;
+            ResourceOperation operationType = ResourceOperation.NoOperation;
+
             if (existingObservation == null)
             {
                 var newObservation = GenerateObservation(config, observationGroup, identifier, ids);
                 result = await _client.CreateAsync(newObservation).ConfigureAwait(false);
-                _logger.LogMetric(IomtMetrics.FhirResourceSaved(ResourceType.Observation, ResourceOperation.Created), 1);
+                operationType = ResourceOperation.Created;
             }
             else
             {
+                Model.Observation mergedObservation = null;

```

Comment:
>mergedObservation [](http://example.com/codeflow?start=34&length=17)

This change + the addition of the operationType is causing the state of the entire of the outer method to be captured in an implicit class by the compiler and passed into the lambdas.

Not the end of the world but I believe we can do something similar below and avoid it.

See:
[https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/lambda-expressions#capture-of-outer-variables-and-variable-scope-in-lambda-expressions](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/lambda-expressions#capture-of-outer-variables-and-variable-scope-in-lambda-expressions)

Reply:
It looks like I forgot about the existingObservation which was outside scope and being implicitly captured already.  If we can eliminate all of them great.  Otherwise should be fine.

Reply:
I was able to easily make some changes to avoid variable capture in most cases, but the "existingObservation" issue is one that is a bit harder. I'm going to punt this one down the road.